bgpd/bgp_mplsvpn.c

```diff
@@ -1493,17 +1514,15 @@ static mpls_label_t bgp_mplsvpn_get_label_for_path_vrf(struct bgp_path_info *pi,
 		return from_bgp->vpn_policy[afi].tovpn_label;
 	}
 
-	/* check that the nexthop reachability is connected
+	/* extract the resolved nexthop given by BNC
+	 * if nexthop type is not connected, then use the resolved nexthop
 	 */
-	for (nhop = bnc->nexthop; nhop; nhop = nhop->next) {
-		if (nhop->type == NEXTHOP_TYPE_IFINDEX)
-			break;
-	}
-	if (!nhop)
-		return from_bgp->vpn_policy[afi].tovpn_label;
-
-	/* the nexthop is in bnc->prefix */
-	nh_to_use_ptr = &bnc->prefix;
```

Comment:
do not remove what you added a few commits ago

Reply:
actually, the change requires to change the behaviour.
all the nexthop types ( except blackholes are eligible for mpls allocation per nexthop).